rq/group.py

```diff
@@ -0,0 +1,124 @@
+from typing import List, Optional
+from uuid import uuid4
+
+from redis import Redis
+from redis.client import Pipeline
+
+from . import Queue
+from .exceptions import NoSuchGroupError
+from .job import Job
+from .queue import EnqueueData
+from .utils import as_text
+
+
+class Group:
+    """A Group is a container for tracking multiple jobs with a single identifier."""
+
+    REDIS_GROUP_NAME_PREFIX = 'rq:group:'
+    REDIS_GROUP_KEY = 'rq:groups'
+
+    def __init__(self, connection: Redis, name: str = None):
+        self.name = name if name else str(uuid4().hex)
+        self.connection = connection
+        self.key = '{0}{1}'.format(self.REDIS_GROUP_NAME_PREFIX, self.name)
+
+    def __repr__(self):
+        return "Group(id={})".format(self.name)
+
+    def _add_jobs(self, jobs: List[Job], pipeline: Pipeline):
+        """Add jobs to the group"""
+        pipe = pipeline if pipeline else self.connection.pipeline()
```

Comment:
This line is no longer needed since `pipeline` is no longer optional.

Reply:
Fixed